---
title: "Applied PCA"
subtitle: "An incomplete tour of example use cases"
author:
  - Elizabeth King
  - Kevin Middleton
format:
  revealjs:
    theme: [default, custom.scss]
    standalone: true
    self-contained: true
    logo: QMLS_Logo.png
    slide-number: true
    show-slide-number: all
bibliography: Multivariate.bib
csl: evolution.csl
---

## Applications of PCA 


```{r}
#| label: setup
#| echo: false
#| warning: false
#| message: false

library(tidyverse)
library(cowplot)
library(readxl)
library(factoextra)
library(palmerpenguins)
ggplot2::theme_set(theme_cowplot())
```

PCA (and other eigen-like analyses) is one of the most useful tools in data science

- Create a new holistic variable 
- Reduce dimensionality
- A way to quantify and describe shared variance
- Separates shared variation into independent sets

## Some Practical Issues in PCA

- Interpretation of the PCs in a biological context
- Deciding on the number of PCs to consider in further analyses


## Penguin PC meanings

```{r}
#| echo: true

penguins_vars <- penguins |>
  dplyr::select(body_mass_g, ends_with("_mm")) |>
  drop_na()

pen_pca <- prcomp(penguins_vars, center = TRUE, scale. = TRUE)

pen_pca

```

## Penguin PC meanings

```{r}

fviz_pca_var(pen_pca, addlabels = TRUE, repel = TRUE)

```


## Which PCs?

```{r}

summary(pen_pca)

```



## Penguin PC meanings

```{r}
p1 <- fviz_pca_var(pen_pca, addlabels = TRUE, repel = TRUE, axes = c(2,3))

p2 <- fviz_pca_var(pen_pca, addlabels = TRUE, repel = TRUE, axes = c(3,4))

plot_grid(p1,p2, ncol=2)
```

## Multivariate Data: Squirrels

```{r}
MM <- read_excel("Data/mammals.xlsx", na = "NA") |>
  filter(family == "Sciuridae" & genus != "Marmota") |>
  dplyr::select(order, family, genus, species, adult_body_mass_g,
                adult_head_body_len_mm, diet_breadth, gestation_len_d,
                habitat_breadth, litter_size,  neonate_body_mass_g,
                sexual_maturity_age_d) |>
  rename(adult_mass = adult_body_mass_g,
         body_length = adult_head_body_len_mm,
         gest_length = gestation_len_d,
         neonate_mass = neonate_body_mass_g,
         maturity_age = sexual_maturity_age_d) |>
  drop_na() 

mod <- lm(maturity_age ~ adult_mass + body_length + diet_breadth +
            gest_length + habitat_breadth + litter_size + neonate_mass,
          data = MM)

summary(mod)
```

## Squirrel PC meanings

```{r}
sqPCA <- prcomp(~ adult_mass + body_length + diet_breadth +
            gest_length + habitat_breadth + litter_size + neonate_mass,
          data = MM, scale. = TRUE, center=TRUE)

sqPCA

```

## Squirrel PC meanings

```{r}

fviz_pca_var(sqPCA, addlabels = TRUE, repel = TRUE)

```


## Which PCs?

```{r}

summary(sqPCA)

```


## Squirrel PC meanings

```{r}
p1 <- fviz_pca_var(sqPCA, addlabels = TRUE, repel = TRUE)

p2 <- fviz_pca_var(sqPCA, addlabels = TRUE, repel = TRUE, axes = c(2,3))

p3 <- fviz_pca_var(sqPCA, addlabels = TRUE, repel = TRUE, axes = c(3,4))

p4 <- fviz_pca_var(sqPCA, addlabels = TRUE, repel = TRUE, axes = c(4,5))

plot_grid(p2,p3,p4, nrow = 1)

```


## Use all?

```{r}
#| echo: true

mod <- lm(MM$maturity_age ~ sqPCA$x)
summary(mod)

```


## Tour of Some Use Cases 

- Creating useful composites
- Creating a useful rotation
- Identifying confounding variables

## Creating a useful composite for analysis

- Often many measures are attempting to quantify one biological characterstic
- Composite variables can be more biologically meaningful than the component measures

## Creating a useful composite for analysis

Example dataset:

- Cultivars of *Brassica oleracea* planted in a common garden
- Measured herbivore species presence and abundance
- Measured glucosinolate concentrations (10 chemicals)

<div class="ref">
Poelman et al [-@Poelman2009-up]
</div>

## Glucosinolate composite

`~ Glu1 + Glu2....Glu10`

- PC1 explained 72% of the variance
- increasing scores on PC1 = increase in glucoiberin and decrease in glucoraphanin

## Glucosinolate composite

![](./Images/gluc1_pca.png){fig-align="center" width=50%}

## Herbivore composite

![](./Images/gluc2.png){fig-align="center" width=80%}

## Creating a useful composite for analysis

:::: {.columns}

::: {.column width="50%"}

Example dataset:

- 32 fish species in a cichlid clade
- morphology & locomotor traits (16 total)
- performed phylogenetic pca

:::

::: {.column width="50%"}

![](./Images/cichlid_phylo.png){fig-align="center" width=100%}
<div class="ref">
Astudillo-Clavjo et al [-Astudillo-Clavijo2015-gv]
</div>

:::

::::

## Creating a useful composite for analysis

![](./Images/fish_pc.jpg){fig-align="center" width=80%}

## PCA as a way to achieve a useful rotation

- PCA can be used to reorganize a dataset 
    - consolidate most of the variance
    - create orthogonal variables
- Use all in following analysis

## PCA as a way to achieve a useful rotation

Example dataset:

- 10 measures of antlers for ~500 male deer
- 38,000 SNPs > genetic relatedness & GWAS

<div class="ref">
Peters et al [-@Peters2022-as]
</div>

## PCA as a way to achieve a useful rotation

![](./Images/antler_pc.png){fig-align="center" width=80%}


<div class="ref">
Peters et al [-@Peters2022-as]
</div>

## PCA as a way to achieve a useful rotation

![](./Images/antler_map.png){fig-align="center" width=80%}

<div class="ref">
Peters et al [-@Peters2022-as]
</div>


## PCA as a way to identify important confounding variables

Batch effects

- Often (but not always) not of interest
- Due to non-independence of sets of samples
- Failure to account for these 
    - at best: lowers power
    - at worst: leads to incorrect conclusions
      
> The signature of batch effects is shared variance

## Known and Unknown Batches

Example dataset:

- Activity levels of growth factor receptor network (GFRN) pathways in relation to breast cancer progression
- Treatment: specific GFRN oncogenes were induced by transfection 
- Controls were transfected with a vector that expresses a green fluorescent protein (GFP)
- Experiment was done in 3 batches. 

<div class="ref">
Zhang et al [-@Zhang2020-va]
</div>


## Known and Unknown Batches

`~ Gene1 + Gene2 + Gene3 ....... Gene17000`

![](./Images/combat_uncor.png){fig-align="center" width=40%}

## Known and Unknown Batches in RNA-seq

![](./Images/combat.png){fig-align="center" width=100%}

## Population Structure

Example dataset:

- Genome sequencing at ~2.8 million SNPs
- 17 tigers from regions in India

<div class="ref">
Khan et al [-@Khan2022-da]
</div>


## Population Structure

`~ SNP1 + SNP2 + SNP3 ....... SNP2800000`

![](./Images/tiger_popst.png){fig-align="center" width=60%}

## Your Data 

- What variables in your datasets share variance?

## References

::: {#refs}
:::